WBooks/src/app/screens/BookDetails/components/BookComments/index.js

```diff
@@ -0,0 +1,41 @@
+import React, { PureComponent } from 'react';
+import { TouchableOpacity, Text } from 'react-native';
+
+import comments from '../../../../../constants/comments';
+
+import CommentItem from './components/CommentItem';
+import styles from './styles';
+
+const INITIAL_SHOWN_COMMENTS = 2;
+
+class BookComments extends PureComponent {
+  constructor(props) {
```

Comment:
replace this constructor defining state directly like:

```
class BookComments extends PureComponent {
    state = { showAll: false }
}
```

another observation is you have to avoid using PureComponent if  have a state because Pure Component try to keep this component without changes but having a state is propense to trigger re renders. So its incompatible. Replace with `Component`